### PR TITLE
Add tests and fix updating of views

### PR DESCRIPTION
**Pull request by akaDuality**

### CODE REVIEW

File: Document/Sources/Document/Drawing/DrawingView.swift [hunk 1]
```diff
@@ -66,8 +66,9 @@ public extension DrawingView {
     }
     
     func removeAll() {
-        for contol in drawnControls {
+        for contol in drawnControls.reversed() {
```
Comment (drucelweisse) [inline Document/Sources/Document/Drawing/DrawingView.swift:69]: typo for control

File: VoiceOver Designer/Features/Sources/Editor/EditorPresenter.swift [hunk 2]
```diff
@@ -22,15 +23,37 @@ public class EditorPresenter {
...
     
-    func draw() {
-        document.controls.forEach { control in
-            drawingController.drawControl(from: control)
+    private var cancellables = Set<AnyCancellable>()
+    
+    private func redrawOnControlChanges() {
+        document
+            .controlsPublisher
+            .sink { controls in
```
Comment (drucelweisse) [inline VoiceOver Designer/Features/Sources/Editor/EditorPresenter.swift:35]: would that create retain? probably should user weak self capture

**Review by drucelweisse — COMMENTED** [review]
Like that extracting to files in Editor tests

**Review by drucelweisse — APPROVED** [review]

File: Document/Sources/Document/Drawing/DrawingView.swift (hunk 1 above)
Comment (akaDuality) [reply Document/Sources/Document/Drawing/DrawingView.swift:69]: Thank you!

Screen Shot 2022-08-06 at 23 40 21

File: VoiceOver Designer/Features/Sources/Editor/EditorPresenter.swift (hunk 2 above)
Comment (akaDuality) [reply VoiceOver Designer/Features/Sources/Editor/EditorPresenter.swift:35]: awesome, we can write to something like 

```
document
    .controlsPublisher
    .sink(receiveValue: redraw)
    .store(in: &cancellables)
```

And avoid memory management.

I'll use this aproach in the next PR, resolve it here. Thank you for your attention to details!

Comment (akaDuality) [reply VoiceOver Designer/Features/Sources/Editor/EditorPresenter.swift:35]: `[More about that](https://www.swiftbysundell.com/articles/combine-self-cancellable-memory-management/)`

**Pull request merged**